pubsub/azure/eventhubs/metadata.yaml

```diff
@@ -0,0 +1,74 @@
+authenticationProfiles:
+  - title: "Connection string"
+    description: "Authenticate using a connection string."
+    metadata:
+      - name: connectionString
+        required: true
+        sensitive: true
+        description: "Connection string for the Event Hub or the Event Hub namespace. Mutally exclusive with eventHubNamespace field."
+        example: '"Endpoint=sb://{EventHubNamespace}.servicebus.windows.net/;SharedAccessKeyName={PolicyName};SharedAccessKey={Key};EntityPath={EventHub}"
+        or "Endpoint=sb://{EventHubNamespace}.servicebus.windows.net/;SharedAccessKeyName={PolicyName};SharedAccessKey={Key}"'
+builtinAuthenticationProfiles:
+  - name: "azuread"
+    metadata:
+      - name: eventHubNamespace
+        description: "The Event Hub Namespace name. Mutually exclusive with connectionString field."
+        example: '"namespace"'
+        required: true
+metadata:
+  - name: storageAccountName
+    description: "Storage account name to use for the checkpoint store."
+    type: string
+    required: true
+    example: "myeventhubstorage"
+  - name: storageAccountKey
+    description: "Storage account key for the checkpoint store account. When using Azure AD, it’s possible to omit this if the service principal has access to the storage account too."
+    type: string
+    sensitive: true
+    example: "112233445566778899"
+  - name: storageConnectionString
+    description: "Connection string for the checkpoint store, alternative to specifying storageAccountKey."
+    type: string
+    sensitive: true
+    example: "DefaultEndpointsProtocol=https;AccountName=myeventhubstorage;AccountKey=<account-key>"
+  - name: storageContainerName
+    description: "Storage container name for the storage account name."
+    type: string
+    required: true
+    example: "myeventhubstoragecontainer"
+  - name: enableEntityManagement
+    description: "Boolean value to allow management of the EventHub namespace and storage account."
+    type: bool
+    default: "false"
+    required: true
+    example: "false"
+
+  # The following four properties are needed only if enableEntityManagement is set to true
+  - name: resourceGroupName
+    description: "Name of the resource group the Event Hub namespace is part of. Required when entity management is enabled."
+    type: string
+    example: "test-rg"
+  - name: subscriptionID
+    description: "Azure subscription ID value. Required when entity management is enabled."
+    type: string
+    example: "azure subscription id"
+  - name: partitionCount
+    description: "Number of partitions for the new Event Hub namespace. Used only when entity management is enabled."
+    type: number
+    default: "1"
+    example: "2"
+  - name: messageRetentionInDays
+    description: "Number of days to retain messages for in the newly created Event Hub namespace. Used only when entity management is enabled."
+    type: number
+    default: "1"
+    example: "90"
```

Comment:
If they require Entity Management and Entity Mangement can only be enabled when using Azure AD, we should move those to `azuread` auth profile. A similar comment was made to my PR on https://github.com/dapr/components-contrib/pull/2784